### ISSUE TITLE
suggestion: add a compatibility flag for the ssg reloading bug

### ISSUE THREAD

**Issue by fraansg** [fabiangreffrath/woof#174]
In MBF (i don't know if also in boom), if you run out of ammo while firing the ssg, the reload animation of the gun is also shown (unlike vanilla), bug that occurs also in Woof!. I would like to know if there is a possibility of incorporating a compatibility flag in Woof!, as was it done with the 3-key door bug.

**Comment by rfomin** [comment]
PrBoom+ fixes this in A_CheckReload()
https://github.com/coelckers/prboom-plus/blob/febe0e16b855b9f893be2268e782596edb984509/prboom2/src/p_pspr.c#L409-L415

Maybe we can fix it in mbf21? @kraflab, could you please mention this in the spec?

**Comment by fabiangreffrath** [comment]
If the bug has been introduced in Boom, why doesn't PrBoom+ fix it for Vanilla complevels?

> I would like to know if there is a possibility of incorporating a compatibility flag in Woof!, as was it done with the 3-key door bug.

This flag has been removed again. The bug is strictly present in MBF but not in Boom, and we now track compatibility levels similar to PrBoom+. Carrying a separate flag for this bug would collide with the flags that PrBoom+ already assigned to other fixes.

**Comment by rfomin** [comment]
> If the bug has been introduced in Boom, why doesn't PrBoom+ fix it for Vanilla complevels?

I think this is a PrBoom + oversight. Also, the bug appears to be purely visual and does not affect demo compatibility.

**Comment by fabiangreffrath** [comment]
If this is an oversight, I don't understand why they added a compat check for *higher* complevels?

**Comment by rfomin** [comment]
PrBoom+ has compatibility checks for visual bugs.

**Comment by hackneyed-one** [comment]
I don't believe it is purely visual and it was introduced in Boom since both Boom 2.02 and MBF show the full reloading animation when 1 or 0 shells remain. Simply put, taking longer (or shorter) to switch weapons will have an affect on demos where engines don't account for it. There is an old comment from chungy about the bug at freedoom. https://github.com/freedoom/freedoom/issues/57

I may be going about this all wrong or something, I barely know bash scripting but anyway... As a test I compiled a version of prboom-plus-master from Apr 26, 2021 without any edits and another one where I change the ">= prboom_4_compatibility" to "<= prboom_4_compatibility" for P_CheckAmmo.

I recorded a Boom demo with the edited executable like this: ./prboom-plus-ssgreload -skill 4 -warp 05 -complevel boom -record ssgreload

My Demo as recorded and played back in the edited executable goes like this: I pick up the SSG and fire 2 shots. I pick up the soul-sphere and kill the chain-gunner on the left. I pick up his chain-gun, turn around and switch back to the SSG. I fire at the wall to force an auto switch back to the chain-gun SKIPPING the SSG reload. Then I shoot the chain-gunner with the chain-gun until I realize he is dead and stop firing. I quickly turn left and fire all my bullets into the wall and quit while still alive. 

However playing my demo in the non-edited executable or Boom 2.02 in Dosbox is exactly the same until the auto switch where the full reload animation happens. I begin turning away from the 2nd chain-gunner before I've even started shooting him and he kills me while I am shooting at the wall ending with 34 bullets and a dead Doomguy.

As much as I would like to see this fixed I think it has to wait for the new mbf+/mbf21 standard I've been hearing about if demo compatibility is to be preserved for any Boom and MBF/complevel 11.

[ssgreload.zip](https://github.com/fabiangreffrath/woof/files/6415740/ssgreload.zip)
